### PR TITLE
Add debug logs for the URI syntax and URI pattern syntax issues

### PR DESCRIPTION
**Pull request by sadilchamishka**
### Proposed changes in this pull request

$subject

Related to https://github.com/wso2/product-is/issues/15696

### CODE REVIEW

File: components/org.wso2.carbon.identity.auth.valve/src/main/java/org/wso2/carbon/identity/auth/valve/AuthenticationValve.java [hunk 1]
```diff
@@ -167,7 +167,11 @@ public void invoke(Request request, Response response) throws IOException, Servl
             log.error("Identity Runtime Exception occurred in Authentication valve :", e);
             APIErrorResponseHandler.handleErrorResponse(authenticationContext, response,
                     HttpServletResponse.SC_SERVICE_UNAVAILABLE, null);
-        } catch (URISyntaxException | PatternSyntaxException e) {
+        } catch (URISyntaxException e) {
+            log.debug("Invalid URI syntax of the request :", e);
```
Comment (sachiniWettasinghe) [inline components/org.wso2.carbon.identity.auth.valve/src/main/java/org/wso2/carbon/identity/auth/valve/AuthenticationValve.java:171]:
```suggestion
            log.debug("Invalid URI syntax of the request: ", e);
```

Comment (sachiniWettasinghe) [reply components/org.wso2.carbon.identity.auth.valve/src/main/java/org/wso2/carbon/identity/auth/valve/AuthenticationValve.java:171]: Also check for isDebugEnabled

File: components/org.wso2.carbon.identity.auth.valve/src/main/java/org/wso2/carbon/identity/auth/valve/AuthenticationValve.java [hunk 2]
```diff
@@ -167,7 +167,11 @@ public void invoke(Request request, Response response) throws IOException, Servl
...
+            APIErrorResponseHandler.handleErrorResponse(null, response, HttpServletResponse.SC_BAD_REQUEST, null);
+        } catch (PatternSyntaxException e) {
+            log.debug("Invalid pattern syntax of the request :", e);
```
Comment (sachiniWettasinghe) [inline components/org.wso2.carbon.identity.auth.valve/src/main/java/org/wso2/carbon/identity/auth/valve/AuthenticationValve.java:174]:
```suggestion
            log.debug("Invalid pattern syntax of the request: ", e);
```

File: components/org.wso2.carbon.identity.auth.valve/src/main/java/org/wso2/carbon/identity/auth/valve/AuthenticationValve.java (hunk 1 above)
Comment (sadilchamishka) [reply components/org.wso2.carbon.identity.auth.valve/src/main/java/org/wso2/carbon/identity/auth/valve/AuthenticationValve.java]: In this case we may not need as we don't do any costly string concatenation etc..
https://www.igorkromin.net/index.php/2015/05/12/are-guard-statements-like-isdebugenabled-necessary-when-using-log4j/

File: components/org.wso2.carbon.identity.auth.valve/src/main/java/org/wso2/carbon/identity/auth/valve/AuthenticationValve.java (hunk 2 above)
Comment (sadilchamishka) [reply components/org.wso2.carbon.identity.auth.valve/src/main/java/org/wso2/carbon/identity/auth/valve/AuthenticationValve.java]: Fixed by - https://github.com/wso2-extensions/identity-carbon-auth-rest/pull/236/commits/4cce13a083b11da0866684a6d30e1edc095651f1

**Review by sachiniWettasinghe — APPROVED** [review]

**Pull request merged**